pkg/api/mlflow/dao/repositories/metric.go

```diff
@@ -154,7 +154,7 @@ func (r MetricRepository) GetMetricHistories(
 		query := r.db.WithContext(ctx).Model(
 			&database.Run{},
 		).Joins(
-			"LEFT JOIN experiments ON experiments.experiment_id = runs.experiment_id",
+			"INNER JOIN experiments ON experiments.experiment_id = runs.experiment_id",
 		).Where(
 			"experiments.namespace_id = ?", namespaceID,
```

Comment:
this should be replaced by the same kind of inner join we do for apps, so that we filter whilst joining instead of joining everything and then having the where clause applied — if we go for the helper described in https://github.com/G-Research/fasttrackml/pull/399/files#r1358038830, this can all be made simpler and more readable IMHO